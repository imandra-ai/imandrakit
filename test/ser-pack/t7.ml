type 'a foo = {
  x: 'a option;
  y: (int * 'a) list;
}
[@@deriving serpack, show]

type ('a, 'b) bar = {
  f1: 'a foo list;
  f2: 'b foo list;
}
[@@deriving serpack, show]

type bar2 = (int, bool) bar [@@deriving show, serpack]

let b : bar2 =
  {
    f1 = [ { x = Some 1; y = [ 0, 0; 1, 1; 2, 2 ] } ];
    f2 = [ { x = None; y = [] }; { x = Some true; y = [ 0, true; 1, false ] } ];
  }
;;

Format.printf "b: %a@." pp_bar2 b

let s = Imandrakit_ser_pack.to_cbor_string bar2_to_serpack b;;

Format.printf "len=%d@." (String.length s)

let b' = Imandrakit_ser_pack.of_cbor_string_exn bar2_of_serpack s;;

Format.printf "b': %a@." pp_bar2 b';;
assert (b = b')
