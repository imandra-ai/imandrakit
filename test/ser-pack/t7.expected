b: { T7.f1 = [{ T7.x = (Some 1); y = [(0, 0); (1, 1); (2, 2)] }];
     f2 =
     [{ T7.x = None; y = [] };
       { T7.x = (Some true); y = [(0, true); (1, false)] }]
     }
len=51
b': { T7.f1 = [{ T7.x = (Some 1); y = [(0, 0); (1, 1); (2, 2)] }];
      f2 =
      [{ T7.x = None; y = [] };
        { T7.x = (Some true); y = [(0, true); (1, false)] }]
      }
